qek/data/extractors.py

```diff
@@ -778,6 +829,134 @@ def run(self) -> CloudExtracted:
+            your computer while the QPU was executing your sequences.
+    """
+
+    def __init__(
+        self,
+        compiler: BaseGraphCompiler[GraphType],
+        project_id: str,
+        username: str,
+        device_name: str = "FRESNEL",
+        password: str | None = None,
+        batch_ids: list[str] | None = None,
+        path: Path | None = None,
+    ):
+        super().__init__(
+            compiler=compiler,
+            project_id=project_id,
+            username=username,
+            device_name=device_name,
+            password=password,
+            batch_ids=batch_ids,
+            path=path,
+        )
+
+    def run(self) -> PasqalCloudExtracted:
+        return self._run(emulator=None, config=None, state_extractor=lambda job, _seq: job.result)
+
+
+class RemoteEmuMPSExtractor(BaseRemoteExtractor[GraphType]):
+    """
+    An Extractor that uses a remote high-performance emulator (EmuMPS)
+    published on Pasqal Cloud, to run sequences compiled from graphs.
+
+    Performance note (servers and interactive applications only):
+        If your code is meant to be executed as part of an interactive application or
+        a server, you should consider calling `await extracted` before your first call
+        to any of the methods of `extracted`. Otherwise, you will block the main thread.
+
+        If you are running this as part of an experiment, a Jupyter notebook, etc. you
+        may ignore this performance note.
+
+    Args:
+        path: Path to store the result of the run, for future uses.
+            To reload the result of a previous run, use `LoadExtractor`.
+        project_id: The ID of the project on the Pasqal Cloud API.
+        username: Your username on the Pasqal Cloud API.
+        password: Your password on the Pasqal Cloud API. If you leave
+            this to None, you will need to enter your password manually.
+        device_name: The name of the device to use. As of this writing,
+            the default value of "FRESNEL" represents the latest QPU
+            available through the Pasqal Cloud API.
+        batch_id: Use this to resume a workflow e.g. after turning off
+            your computer while the QPU was executing your sequences.
+    """
+
+    def __init__(
+        self,
+        compiler: BaseGraphCompiler[GraphType],
+        project_id: str,
+        username: str,
+        device_name: str = "FRESNEL",
+        password: str | None = None,
+        batch_ids: list[str] | None = None,
+        path: Path | None = None,
+    ):
+        super().__init__(
+            compiler=compiler,
+            project_id=project_id,
+            username=username,
+            device_name=device_name,
+            password=password,
+            batch_ids=batch_ids,
+            path=path,
+        )
+
+    def run(self, dt: int = 10) -> PasqalCloudExtracted:
+        def extractor(job: Job, sequence: pl.Sequence) -> dict[str, int] | None:
```

Comment:
Does this callback has to return `None` ? Wouldn't an empty `dict` suffice ?

Reply:
Mmmmh... The `None` case means that the job somehow didn't return a result, while the empty `dict` means that the job returned an empty result. In practice, neither should happen, but just in case, I don't see any major reason to mask the difference.

Reply:
Well, if it didn't return a result, it means there has been an issue somewhere. In that case, an exception should have been raised. IMHO, `None` creates more confusion because it is neither a result nor an exception.